src/mixins/allowable-actions-mixin.ts

```diff
@@ -2,101 +2,155 @@ import Vue from 'vue'
 import { Component } from 'vue-property-decorator'
 import { Getter } from 'vuex-class'
 import { getFeatureFlag } from '@/utils'
-import { AllowableActions, CorpTypeCd, Routes } from '@/enums'
+import { AllowableActions, CorpTypeCd, FilingTypes, RestorationTypes, Routes } from '@/enums'
+import { AllowedActionsIF } from '@/interfaces'
 
 @Component({})
 export default class AllowableActionsMixin extends Vue {
+  @Getter getAllowedActions!: AllowedActionsIF
   @Getter getLegalType!: CorpTypeCd
-  @Getter hasBlocker!: boolean
-  @Getter hasBlockerExceptStaffApproval!: boolean
   @Getter isBComp!: boolean
-  @Getter isBenBcCccUlc!: boolean
   @Getter isCoop!: boolean
+  @Getter isFirm!: boolean
   @Getter isGoodStanding!: boolean
-  @Getter isHistorical!: boolean
-  @Getter isPartnership!: boolean
   @Getter isRoleStaff!: boolean
-  @Getter isSoleProp!: boolean
 
   /**
    * Returns True if the specified action is allowed, else False.
    * @param action the action to check
    */
   isAllowed (action: AllowableActions): boolean {
-    const businessId = sessionStorage.getItem('BUSINESS_ID')
+    const isBusiness = !!sessionStorage.getItem('BUSINESS_ID') // ie, not a temporary business
 
     switch (action) {
-      case AllowableActions.ADD_DETAIL_COMMENT: {
-        return (!!businessId && this.isRoleStaff)
+      case AllowableActions.ADDRESS_CHANGE: {
+        if (this.isFirm) {
+          return this.isAllowedFiling(FilingTypes.CHANGE_OF_REGISTRATION)
+        }
+        return this.isAllowedFiling(FilingTypes.CHANGE_OF_ADDRESS)
       }
 
-      case AllowableActions.ADD_STAFF_COMMENT: {
-        return this.isRoleStaff
+      case AllowableActions.ADMINISTRATIVE_DISSOLUTION: {
+        // FUTURE: check dissolution type
```

Comment:
Dissolution type is not yet provided by the API - see ticket 15548. There is a note in that ticket to also patch this code and Voluntary Dissolution below, but the current code works as-is.